packages/transporter/src/concerns/retryableRequest.ts

```diff
@@ -40,6 +40,8 @@ export function retryableRequest<TResponse>(
     request.method !== MethodEnum.Get
       ? {}
       : {
+          // if AuthMode.WithinData, we forcibly map apiKey back to a query param
+          'x-algolia-api-key': transporter.data?.apiKey,
```

Comment:
this might be implemented wrong?

Reply:
Not sure what you mean by "wrong", but I have a extra question: is the idea here to move auth of all non GET requests to the body?